src/util/observatory.ts

```diff
@@ -0,0 +1,63 @@
+import {BoaviztaCpuImpactModel} from '../lib';
+
+/**
+ * Pipeline for computing impacts based on observation.
+ */
+export class Observatory {
+  private observations: any;
+
+  /**
+   * Init observations object.
+   */
+  constructor(observations: any) {
+    this.observations = observations;
+  }
+
+  /**
+   * Apply `boavizta` model computation to observations.
+   */
+  private async boaviztaHandler(params: any) {
+    const modelInstance = await new BoaviztaCpuImpactModel().configure(
+      'test',
+      params
+    );
+
+    const preparedObservations = this.observations.map((observation: any) => ({
```

Comment:
There does need to be some manipulation of data in a general sense but we should not have to hard code specific transformations for specific models. The data is either passed in a way that the model can take without any transformation, or if some transformation is required the model is responsible for doing it itself.

Reply:
Agree, I don't like this either. It's better to tune yaml/model to use the same interface rather than mapping to one interface, then mapping the model result back to original pattern